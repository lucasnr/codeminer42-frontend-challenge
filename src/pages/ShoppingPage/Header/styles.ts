import styled from 'styled-components';

export const Wrapper = styled.header`
	background-color: ${({ theme }) => theme.colors.secondary};
	color: ${({ theme }) => theme.text.secondary};
`;

export const Container = styled.div`
	align-items: center;
	display: flex;
	justify-content: space-between;
	margin: 0px auto;
	max-width: 62rem;
	padding: 1rem;
	width: 100%;
`;

export const Title = styled.h1`
	font-size: 2rem;
	font-weight: 700;
`;

export const User = styled.div`
	align-items: center;
	display: flex;
	font-size: 1.125rem;
	font-weight: 500;

	img {
		border: 0.125rem solid ${({ theme }) => theme.text.secondary};
		border-radius: 50%;
		margin-right: 1rem;
		height: 2.75rem;
		width: 2.75rem;
	}
`;
