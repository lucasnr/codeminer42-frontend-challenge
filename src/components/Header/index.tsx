import React from 'react';

import { Wrapper, Container, Title, User } from './styles';

import avatar from '~/assets/img/diverseui-male-67.jpg';

const Header: React.FC = () => {
	return (
		<Wrapper>
			<Container>
				<Title>Shopping</Title>
				<User>
					<img alt="User's avatar" src={avatar} />
					<span>John Doe</span>
				</User>
			</Container>
		</Wrapper>
	);
};

export default Header;
