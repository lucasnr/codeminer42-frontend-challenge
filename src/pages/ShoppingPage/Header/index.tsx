import React from 'react';

import { Wrapper, Container, Title, User } from './styles';

const Header: React.FC = () => {
	return (
		<Wrapper>
			<Container>
				<Title>Shopping</Title>
				<User>
					<img
						alt="User's avatar"
						src="https://static.diverseui.com/male-67.jpg"
					/>
					<span>John Doe</span>
				</User>
			</Container>
		</Wrapper>
	);
};

export default Header;
