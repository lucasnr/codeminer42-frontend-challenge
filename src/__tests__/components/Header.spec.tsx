import React from 'react';
import { shallow } from 'enzyme';

import Header from '~/components/Header';

describe('Header component', () => {
	it('renders without fail and matches snapshot', () => {
		const wrapper = shallow(<Header />);
		expect(wrapper).toMatchSnapshot();
	});
});
