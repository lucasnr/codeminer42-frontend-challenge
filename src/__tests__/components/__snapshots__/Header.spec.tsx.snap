// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> renders without fail and matches snapshot 1`] = `
<styled.header>
  <styled.div>
    <styled.h1>
      Shopping
    </styled.h1>
    <styled.div>
      <img
        alt="User's avatar"
        src="diverseui-male-67.jpg"
      />
      <span>
        John Doe
      </span>
    </styled.div>
  </styled.div>
</styled.header>
`;
